core/endorser/endorser.go

```diff
@@ -500,3 +507,48 @@ func CreateCCEventBytes(ccevent *pb.ChaincodeEvent) ([]byte, error) {
 func decorateLogger(logger *flogging.FabricLogger, txParams *ccprovider.TransactionParams) *flogging.FabricLogger {
 	return logger.With("channel", txParams.ChannelID, "txID", shorttxid(txParams.TxID))
 }
+
+// Using the simulation results, build the ChaincodeInterest structure that the client can pass to the discovery service
+// to get the correct endorsement policy for the chaincode(s) and any collections encountered.
+func buildChaincodeInterest(simResult *ledger.TxSimulationResults) (*pb.ChaincodeInterest, error) {
+	ccInterest := &pb.ChaincodeInterest{}
+
+	chaincodes := map[string]struct{}{}
+	if simResult.PvtSimulationResults != nil {
+		// prototype - list collections that are read from (in order to populate NoPublicReads in ChaincodeInterest)
+		for _, prws := range simResult.PvtSimulationResults.GetNsPvtRwset() {
+			for _, cprws := range prws.GetCollectionPvtRwset() {
+				ccInterest.Chaincodes = append(ccInterest.Chaincodes, &pb.ChaincodeCall{
+					Name:            prws.GetNamespace(),
+					CollectionNames: []string{cprws.GetCollectionName()},
+					NoPrivateReads:  !simResult.PrivateReads.Read(prws.Namespace, cprws.CollectionName),
```

Comment:
Just a clarification question - so multiple entries for a single chaincode are expected - one for each collection? I figured that though there a slice for specifying multiple collection names but only single flag for `NoPrivateReads` - so is this a workaround to overcome that?

Reply:
Yes exactly.  I've added a comment in the code to this effect